### PR TITLE
Carbon 2 compatibility

### PR DESCRIPTION
**Pull request by cmorbitzer**
This PR adds compatibility for the `CarbonImmutable` object, introduced in Carbon 2 and available in Laravel 5.8.

### CODE REVIEW

File: tests/TestCase.php [hunk 1]
```diff
@@ -30,21 +32,16 @@ protected function getPackageProviders($app)
 
     protected function setNow($year, int $month = 1, int $day = 1)
     {
-        $newNow = $year instanceof Carbon
-            ? $year
+        $newNow = ($year instanceof Carbon || $year instanceof CarbonInterface)
+            ? $year->copy()
```
Comment (freekmurze) [inline tests/TestCase.php]: Nice!

File: tests/TestCase.php [hunk 2]
```diff
@@ -30,21 +32,16 @@ protected function getPackageProviders($app)
 
     protected function setNow($year, int $month = 1, int $day = 1)
     {
-        $newNow = $year instanceof Carbon
-            ? $year
+        $newNow = ($year instanceof Carbon || $year instanceof CarbonInterface)
```
Comment (freekmurze) [inline tests/TestCase.php]: Do we still need to check for `Carbon` if we already check for `CarbonInterface`?

Comment (cmorbitzer) [reply tests/TestCase.php]: If we want to support Carbon 1, yes

**Comment by cmorbitzer** [comment]
> Could we make this simpler if we just drop support for Carbon 1?

Mildly so, but it would be breaking change. Laravel 5.7 doesn't support Carbon 2 and Laravel 5.8 supports both.

**Comment by freekmurze** [comment]
Let's just drop Carbon v1. This isn't a breaking change. Composer will not pull it into projects where Carbon v2 can't be used. No apps will break.

**Comment by cmorbitzer** [comment]
@freekmurze Done!

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**